app/views/tasks/show.html.erb

```diff
@@ -31,6 +32,46 @@
   <p>(<%= @task.latitude %>, <%= @task.longitude %>)</p>
 <% end %>
 
+<% if @task.latitude.present? and @task.longitude.present? %>
+  <div id="map" class="map" style="height:250px"></div>
+  <div style="display: none;">
+  <a class="overlay" id="task" target="_blank" href=""><%= @task.title %></a>
+  <div id="marker" title="Marker"><%= image_tag "location.png" %></div>
+</div>
+  <script type="text/javascript">
+  var popup = new ol.Overlay({
+    element: document.getElementById('popup')
+  });
+  var map = new ol.Map({
+    target: 'map',
+    layers: [
+      new ol.layer.Tile({
+        source: new ol.source.OSM() 
+      })
+    ],
+    view: new ol.View({
+      center: ol.proj.fromLonLat([<%= @task.longitude %>, <%= @task.latitude %>]),
+      zoom: 16
+    })
+  });
+
+  var pos = ol.proj.fromLonLat([<%= @task.longitude %>, <%= @task.latitude %>]);
+
+  var marker = new ol.Overlay({
+    position: pos,
+    element: document.getElementById('marker'),
+    stopEvent: false
+  });
+  map.addOverlay(marker);
+
+  var task = new ol.Overlay({
+    position: pos,
+    element: document.getElementById('task')
+  });
+  map.addOverlay(task);
```

Comment:
What is the difference between the task overlay and the marker overlay ? Why are there two ?

Reply:
the task overlay is the title of the event shown on the map and the marker overlay is the marker image

Reply:
Let me see if I can condense them into one overlay

Reply:
Thanks ! Interesting, the task title isn't handled by line 38 ?

Reply:
The task title is handled by line 38 as well as lines 67-71. The marker icon is handled on 39 and 60-65.

Reply:
Our comments got mingled. I don't see a value to combining them into one layer, and could see disadvantages, like turning off the titles.

Reply:
Okay got it. Is there anything more for this commit then?

Reply:
I'd like someone more competent in JS to take a look...

Reply:
@marshall-alex @kgf Hey guys, I will take a look at this tomorrow, I was very busy today and was not able to code review.

Reply:
That's great, thanks for your help!

Reply:
@marshall-alex @kgf After reviewing this, I don't see any major issues with the way this is implemented I just made a few comments with questions.